### PR TITLE
Removed prometheus and certmanager crd's

### PR DESCRIPTION
**Pull request by hprotzek**
Removed the obsolete Prometheus crd and replaced cert-manager crd copy with optional helm chart dependency.

### CODE REVIEW

**Comment by wonderflow** [comment]
This PR is great!

Please add `helm dependency update` into the installation guide and our e2e CI system.

**Comment by hprotzek** [comment]
@wonderflow Unfortunately this was not as straightforward as I thought. When using the `cert-manager` helm chart as dependency, the crd's are still missing, when `kubevela` tries to create the `Issuer` and it fails.
It seems, that's because the crd's from `cert-manager` are provided by normal helm templates instead in the crds folder.

We could just addthe manual install of the certmanager crds to the documentation? Or automate that with a Job and a helm `pre-install` hook, to install the certmanager crds?

**Comment by wonderflow** [comment]
> Or automate that with a Job and a helm pre-install hook, to install the certmanager crds?

Is that possible? this would be the best solution. If one user try install kubevela helm chart with webhook enabled, we should check there MUST have a cert-manager installed. If not, we should fail the installation.

**Comment by hprotzek** [comment]
Unfortunately the idea with the helm pre-install hook didn't work neither. It feels all my changes made it worse, so I will close this PR, maybe there are better solutions in the future.

**Comment by wonderflow** [comment]
@hprotzek  It seems this PR is very important as we still conflict with the cluster which has the existing cert-manager CRD installed? 

> that's because the crd's from cert-manager are provided by normal helm templates instead in the crds folder.

Can I ask why ？ If the cert-manager is installed, the crd in its template will still be there and the vela-core shouldn't fail?

**Comment by hprotzek** [comment]
@wonderflow The problem is, when cert-manager is not installed and helm tries to install it as a dependency, the cert-manager crd's are not available when kubevela tries to render the Issuer and Certificate resources. 

I had a look at the prometheus operator helm chart, as they also use cert-manager for the webhook. There they don't install cert-manager as a helm dependency, you need to install it before manually. But they are using [kube-webhook-certgen](https://hub.docker.com/r/jettech/kube-webhook-certgen) as a default (if cert-manager is not installed) to generate the self signed certs. We could use the same mechanism also for kubevela?

**Comment by wonderflow** [comment]
@hprotzek  Sure I think that works for KubeVela.

**Comment by wonderflow** [comment]
issue refer to #1035

**Comment by hprotzek** [comment]
sorry it took so long, I will look into this today :-)

**Comment by hprotzek** [comment]
@wonderflow I have now a working version without the helm crd's from Prometheus and Certmanager. 

If no Certmanager is installed, a helm hook will generate and setup a self signed cert (https://github.com/jet/kube-webhook-certgen). 

If the Certmanager is installed, then the old setup is used, but the Certmanager must be installed manually before Kubevela. 

But during the e2e tests I saw, that the cli is using the certmanager to create certs. Not sure about this functionality and if then the version without Certmanager makes sense at all?

File: Makefile
Comment (wonderflow) [inline Makefile:128]: why use `helm upgrade` instead of `helm install`

File: Makefile
Comment (wonderflow) [inline Makefile:127]: I think we can use the `self signed cert` version to do e2e-test, will that work?

File: charts/vela-core/templates/kubevela-controller.yaml
Comment (wonderflow) [inline charts/vela-core/templates/kubevela-controller.yaml:145]: why remove the mountPath helm value and use a hardcode one

File: docs/en/install.md
Comment (wonderflow) [inline docs/en/install.md:96]: The above description about the default case also need to update , please describe the new mechanism a bit

**Review by wonderflow — COMMENTED** [review]
This is a great PR, some minor comments left before merge

**Comment by wonderflow** [comment]
> @wonderflow I have now a working version without the helm crd's from Prometheus and Certmanager.
> 
> If no Certmanager is installed, a helm hook will generate and setup a self signed cert (https://github.com/jet/kube-webhook-certgen).
> 
> If the Certmanager is installed, then the old setup is used, but the Certmanager must be installed manually before Kubevela.
> 
> But during the e2e tests I saw, that the cli is using the certmanager to create certs. Not sure about this functionality and if then the version without Certmanager makes sense at all?

You can change the e2e test to use the `self signed certificate` instead of cert-manager

File: Makefile
Comment (hprotzek) [reply Makefile:128]: Sorry for the late reply, was on vacation :-) Running helm upgrade with the option --install will do a normal install if the release is missing, if it's already existing, it will upgrade it. This comes handy when you test and run in locally and also works fine in a fresh ci env.

File: Makefile
Comment (hprotzek) [reply Makefile:127]: I tried, but one of the e2e tests failed, when creating an environment and it tries to generate a certificate, expecting an existing `Issuer`.

File: charts/vela-core/templates/kubevela-controller.yaml
Comment (hprotzek) [reply charts/vela-core/templates/kubevela-controller.yaml:145]: I didn't see any case, to set the mount path manually, but I can re-add it again.

File: Makefile
Comment (wonderflow) [reply Makefile:128]: cool, that's good

File: Makefile
Comment (wonderflow) [reply Makefile:127]: OK, maybe we can refactor that in the future

File: charts/vela-core/templates/kubevela-controller.yaml
Comment (wonderflow) [reply charts/vela-core/templates/kubevela-controller.yaml:145]: thanks, please add it back,

**Review by wonderflow — APPROVED** [review]
Thanks for your great work,  LGTM after the doc updated

File: charts/vela-core/templates/kubevela-controller.yaml
Comment (ryanzhang-oss) [inline charts/vela-core/templates/kubevela-controller.yaml:113]: why hard code this?

Comment (wonderflow) [reply charts/vela-core/templates/kubevela-controller.yaml:113]: yes, please add back, thanks, we need it in inner use case

**Comment by wonderflow** [comment]
@hprotzek  Hi, can you help fix the final comment so we can merge this PR, thanks

**Comment by wonderflow** [comment]
Great Job

**Pull request merged**
